docs/further_useful_features.rst

```diff
@@ -0,0 +1,459 @@
+
+
+Secondary Indices
+-----------------
+
+The ability to build and maintain 'secondary' indices are an additional ability
+provided by ``kartothek``. In general, an index is a data structure used to improve
+the speed of read queries. In the context of ``kartothek`` an index is a data structure
+that contains a mapping of every unique value of a given column to references to all the
+partitions where this value occurs.
+
+Explicitly partitioning on a column internally creates a :class:`~kartothek.core.index.PartitionIndex`.
+While this index has a one-to-one mapping of column values to partition references,
+secondary indices have the advantage of being able to contain one-to-many mappings of
+column values to partition references.
+
+.. note::
+
+    The examples we've looked at so far have mostly used functions from the ``eager``
+    backend. As noted earlier, the ``iter`` backend executes operations on the dataset
+    on a per-partition basis and accordingly data inputs are expected to be generators.
+    Although using other iterables such as lists also works, doing so is counter
+    to the intent of the ``iter`` backend (lists would be appropriate in ``eager``).
+
+
+Writing a dataset with a secondary index:
+
+.. ipython:: python
+
+    from kartothek.io.iter import store_dataframes_as_dataset__iter
+
+    # "Generate" 5 dataframes
+    df_gen = (
+        pd.DataFrame({"date": pd.Timestamp(f"2020-01-0{i}"), "X": np.random.choice(10, 10)})
+        for i in range(1, 6)
+    )
+
+    dm = store_dataframes_as_dataset__iter(
+        df_gen,
+        store_factory,
+        "secondarily_indexed",
+        partition_on="date",
+        secondary_indices="X",
+    )
+    dm
+
+    dm = dm.load_all_indices(store_factory())
+    dm.indices["X"].eval_operator("==", 0)  # Show files where `X == 0`
+
+
+As can be seen from the example above, both ``partition_on`` and ``secondary_indices``
+can be specified together. Multiple ``secondary_indices`` can also be added as a list of
+strings.
+
+In general, secondary indices behave like explicit partitions in terms of when and how they
+can and cannot be created. However, when using ``partition_on`` the order of the columns
+provided is important, whereas it is ignored for ``secondary_indices``: this is because the
+underlying processes differ and creation of the :class:`~kartothek.core.index.PartitionIndex`
+is merely a side-effect of using ``partition_on``.
+
+
+Updating an existing dataset
+============================
+
+It's possible to update datasets by adding new physical partitions to them, ``kartothek``
+provides update functions that generally have the prefix `update_dataset` in their names.
+For example, :func:`~kartothek.io.eager.update_dataset_from_dataframes` is the update
+function for the ``eager`` backend.
+
+To see updating in action, let's set up some data and a storage location first and store
+the data there with ``kartothek``:
+
+.. ipython:: python
```

Comment:
Not necessary but I'm curious: Is it possible to arrange this in a collapsable box? After the getting started guide the reader would probably not read the actual data setup and this would help the reading flow. It's useful to be there if somebody want to look up details but in general this is not the feature we want to promote here

Just wondering. A "dont know", "too complicated" or a "will take a lot of time" is a completely satisfying answer at this point :)

Reply:
too complicated